openml/runs/functions.py

```diff
@@ -42,6 +44,19 @@ def run_task(task, model):
     # TODO move this into its onwn module. While it somehow belongs here, it
     # adds quite a lot of functionality which is better suited in other places!
     # TODO why doesn't this accept a flow as input? - this would make this more flexible!
+    flow = sklearn_to_flow(model)
+
+    # returns flow id if the flow exists on the server, -1 otherwise
+    flow_id = flow_exists(flow.name, flow.external_version)
+
+    # skips the run if it already exists and the user opts for this in the config file.
+    # also, if the flow is not present on the server, the check is not needed.
+    if config.avoid_duplicate_runs and flow_id:
+        flow = get_flow(flow_id)
+        setup_id = setup_exists(flow, model)
+        ids = _run_exists(task.task_id, setup_id)
+        if ids:
+            raise PyOpenMLError("Run already exists in server. Run id(s): %s" %str(ids))
```

Comment:
Idk, in some sense the user exhibits unwanted behavior, i.e., he explicitly states that he doesn't want to have duplicate runs on the server, yet he didn't (correctly) check  whether the run is already on the server. This should be reported to the user.

Furthermore, as in this case the variables are static, I see no problem in raising the error.